mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/resourcecluster/ResourceClusterActorClusterUsageTest.java

```diff
@@ -0,0 +1,292 @@
+    }
+
+    @Before
+    public void setupActor() {
+        final Props props =
+            ResourceClusterActor.props(
+                CLUSTER_ID,
+                heartbeatTimeout,
+                assignmentTimeout,
+                checkForDisabledExecutorsInterval,
+                Clock.systemDefaultZone(),
+                rpcService,
+                mantisJobStore,
+                jobMessageRouter,
+                0,
+                "",
+                false,
+                ImmutableMap.of("jdk", "8"));
+
+        resourceClusterActor = actorSystem.actorOf(props);
+        resourceCluster =
+            new ResourceClusterAkkaImpl(
+                resourceClusterActor,
+                Duration.ofSeconds(1),
+                CLUSTER_ID,
+                new LongDynamicProperty(propertiesLoader, "rate.limite.perSec", 10000L));
+    }
+
+    @Test
+    public void testGetTaskExecutorsUsage_WithAllocationAttributes() throws Exception {
+        registerTEs();
+
+        TestKit probe = new TestKit(actorSystem);
+        resourceClusterActor.tell(new GetClusterUsageRequest(CLUSTER_ID, ResourceClusterScalerActor.groupKeyFromTaskExecutorDefinitionIdFunc),
+            probe.getRef());
+        GetClusterUsageResponse usageRes = probe.expectMsgClass(GetClusterUsageResponse.class);
+
+        // The 3 registrations belong to 3 different skus
+        assertEquals(3, usageRes.getUsages().size());
+        for (ContainerSkuID skuID : ImmutableList.of(CONTAINER_DEF_ID_1, CONTAINER_DEF_ID_2, CONTAINER_DEF_ID_3)) {
+            // no scheduling happened so far, so each Te is reported as idle
+            assertIdleAndTotalCount(usageRes, skuID.getResourceID(), 1, 1);
+        }
+
+        // reserve jdk 17 TE and check usage
+        Set<TaskExecutorAllocationRequest> requests = Collections.singleton(TaskExecutorAllocationRequest.of(WorkerId.fromIdUnsafe("late-sine-function-tutorial-1-worker-0-1"), SchedulingConstraints.of(MACHINE_DEFINITION_2, ImmutableMap.of("jdk", "17")), null, 0));
+        assertEquals(
+            TASK_EXECUTOR_ID_3,
+            resourceCluster.getTaskExecutorsFor(requests).get().values().stream().findFirst().get());
+
+        probe = new TestKit(actorSystem);
+        resourceClusterActor.tell(new GetClusterUsageRequest(CLUSTER_ID, ResourceClusterScalerActor.groupKeyFromTaskExecutorDefinitionIdFunc),
+            probe.getRef());
+        usageRes = probe.expectMsgClass(GetClusterUsageResponse.class);
+
+        // sku1 & sku2 -> 1 idle and 1 total. sku3 -> no idle, 1 total.
+        for (ContainerSkuID skuID : ImmutableList.of(CONTAINER_DEF_ID_1, CONTAINER_DEF_ID_2)) {
+            assertIdleAndTotalCount(usageRes, skuID.getResourceID(), 1, 1);
+        }
+        assertIdleAndTotalCount(usageRes, CONTAINER_DEF_ID_3.getResourceID(), 0, 1);
+    }
+
+    @Test
+    public void testGetTaskExecutorsUsage_WithAllocationAttributesWithPendingJobSingleStage() throws Exception {
+        registerTEs();
```

Comment:
nit: maybe call this in setup since its being called in every test.

Reply:
good catch.